src/pool/spawn.rs

```diff
@@ -56,13 +56,29 @@ impl<T> QueueCore<T> {
     /// the action.
     pub fn ensure_workers(&self, source: usize) {
         let cnt = self.active_workers.load(Ordering::SeqCst);
-        if (cnt >> WORKER_COUNT_SHIFT) >= self.config.max_thread_count || is_shutdown(cnt) {
+        if (cnt >> WORKER_COUNT_SHIFT) >= self.config.core_thread_count.load(Ordering::SeqCst)
+            || is_shutdown(cnt)
+        {
             return;
         }
 
         let addr = self as *const QueueCore<T> as usize;
+        let mut unparked_once = false;
+
         unsafe {
-            parking_lot_core::unpark_one(addr, |_| UnparkToken(source));
+            parking_lot_core::unpark_filter(
+                addr,
+                |p: ParkToken| {
+                    if !unparked_once && p.0 <= self.config.core_thread_count.load(Ordering::SeqCst)
```

Comment:
If it's scheduled a lot of tasks before start, then all threads will have workload despite the `core_thread_count`, isn't it?